### ISSUE TITLE
DDP/DP training

### ISSUE THREAD

**Issue by Charlie839242** [drprojects/superpoint_transformer#71]
Hi Damien,

Thanks for your great paper and awesome project!
I am not very sure about one point: simply changing the gpu.yaml to ddp.yaml will make the model train in DDP model, is this correct? And is synchronization of batchnorm supported in this case?
Thanks for you reply!

**Comment by drprojects** [comment]
Hi @Charlie839242, thanks for your interest in the project ! 

That is correct, using the `ddp.yaml` config should allow you to do multi-gpu training. For more information, please refer to the documentation from the [lightning-hydra template](https://github.com/ashleve/lightning-hydra-template) and [lightning](https://lightning.ai/docs) on which this project is based. 

Two remarks though:
- our work focuses on efficient and lightweight models. Hence, multi-GPU is not really our prioritized research direction, so **we have not tested it**
- our models use GraphNorm and not BatchNorm. You will have to investigate whether the [GraphNorm implementation](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.norm.GraphNorm.html) we use properly interfaces with Lightning multi-GPU. I have doubts about it... An easy fix would be to replace the GraphNorms by BatchNorms in your model config and train from scratch, at the cost of a (small) drop in performance in our experiments

PS: if you like or use this project, please give it a ⭐, it means a lot to us !

**Comment by Charlie839242** [comment]
Thank a lot for your quick reply! My question is solved.
